backend/internal/ou/store/store.go

```diff
@@ -0,0 +1,447 @@
+/*
+ * Copyright (c) 2025, WSO2 LLC. (http://www.wso2.com).
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+// Package store provides the implementation for organization unit persistence operations.
+package store
+
+import (
+	"fmt"
+
+	"github.com/asgardeo/thunder/internal/ou/model"
+	dbmodel "github.com/asgardeo/thunder/internal/system/database/model"
+	"github.com/asgardeo/thunder/internal/system/database/provider"
+	"github.com/asgardeo/thunder/internal/system/log"
+)
+
+const loggerComponentName = "OrganizationUnitStore"
+
+// GetOrganizationUnitList retrieves all organization units.
+func GetOrganizationUnitList() ([]model.OrganizationUnitBasic, error) {
+	logger := log.GetLogger().With(log.String(log.LoggerKeyComponentName, loggerComponentName))
+
+	dbClient, err := provider.NewDBProvider().GetDBClient("identity")
+	if err != nil {
+		logger.Error("Failed to get database client", log.Error(err))
```

Comment:
We're already logging errors at service layer. Hence we can remove this.
Check other places as well